validate_email/updater.py

```diff
@@ -37,85 +33,70 @@ class BlacklistUpdater(object):
     """
 
     _refresh_when_older_than: int = 5 * 24 * 60 * 60  # 5 days
-    _is_install_time: bool = False
-
-    @property
-    def _etag_filepath(self) -> str:
-        'Return the ETag file path to use.'
-        return ETAG_FILEPATH_INSTALLED \
-            if self._is_install_time else ETAG_FILEPATH_TMP
-
-    @property
-    def _blacklist_filepath(self) -> str:
-        'Return the blacklist file path to use.'
-        return BLACKLIST_FILEPATH_INSTALLED \
-            if self._is_install_time else BLACKLIST_FILEPATH_TMP
 
     def _read_etag(self) -> Optional[str]:
         'Read the etag header from the stored etag file when exists.'
         for path in [ETAG_FILEPATH_TMP, ETAG_FILEPATH_INSTALLED]:
             try:
-                with open(path) as fd:
-                    return fd.read().strip()
+                return path.read_text().strip()
             except FileNotFoundError:
                 pass
 
-    def _write_etag(self, content: str):
-        'Write the etag of the newly received file to the cache.'
-        path = self._etag_filepath
-        LOGGER.debug(msg=f'Storing ETag response into {path}.')
-        with open(path, 'w') as fd:
-            fd.write(content)
-
     @property
     def _is_old(self) -> bool:
         'Return `True` if the locally stored file is old.'
         true_when_older_than = time() - self._refresh_when_older_than
-        try:
-            ctime = BLACKLIST_FILEPATH_TMP.stat().st_ctime
-            if ctime >= true_when_older_than:
-                # Downloaded tmp file is still fresh enough
-                return False
-        except FileNotFoundError:
-            pass
-        try:
-            ctime = BLACKLIST_FILEPATH_INSTALLED.stat().st_ctime
-        except FileNotFoundError:
-            return True
-        return ctime < true_when_older_than
```

Comment:
This here serves a purpose, an edge case.

If the preinstalled file path doesn't exist (for example at install time), it should return `True` so that the updater will start an update. Besides, your updated version does return `None`, which is not a strict `bool` just boolish.

Reply:
Yes, right, there's a "return True" at the end of my version missing. In practice, this should never happen (this code is not run at install time any more), but I absolutely agree that it's better to have it there.